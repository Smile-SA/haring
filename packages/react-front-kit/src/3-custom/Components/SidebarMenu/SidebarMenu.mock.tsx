import type { IMenuItem } from './SidebarMenu';

import { Chat, HouseLine, Star, User } from '@phosphor-icons/react';

export const menu: IMenuItem[] = [
  {
    children: [{ id: 0, label: 'Home' }],
    id: 1,
    label: 'Home',
    leftIcon: <HouseLine />,
  },
  {
    children: [{ id: 2, label: 'Security' }],
    id: 3,
    label: 'Security',
    leftIcon: <Chat />,
  },
  {
    children: [
      { id: 4, label: 'Release' },
      { children: [{ id: 5, label: 'Account' }], id: 6, label: 'Account' },
      { id: 7, label: 'Upcoming release' },
    ],
    id: 8,
    label: 'Dashboard',
    leftIcon: <Star />,
  },
  { id: 9, label: 'Open Issues', leftIcon: <HouseLine /> },
  {
    children: [
      {
        children: [
          { id: 10, label: 'Wiki pages' },
          { id: 11, label: 'Settings' },
        ],
        id: 12,
        label: 'Dashboard',
      },
      { id: 13, label: 'Home' },
    ],
    id: 14,
    label: 'Pull Requests',
    leftIcon: <User />,
  },
];
