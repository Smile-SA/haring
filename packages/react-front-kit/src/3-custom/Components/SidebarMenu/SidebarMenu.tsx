'use client';

import type { ReactNode } from 'react';

import { Paper } from '@mantine/core';
import { useState } from 'react';

import { CollapseButton } from '../CollapseButton/CollapseButton';

export interface IMenuItem {
  children?: IMenuItem[];
  id: string | number;
  label: string | number;
  leftIcon?: ReactNode;
}

interface ISidebarMenuProps {
  menu: IMenuItem[];
}

function getRecursiveMenu(
  setSelectedId: (id?: string | number) => void,
  selectedId?: string | number,
  menu?: IMenuItem[],
  level = 0
): ReactNode[] | null {
  if (!menu) {
    return null;
  }
  return menu.map(({ children, ...props }, i) => (
    <CollapseButton
      {...props}
      key={props.id ?? i}
      level={level}
      line={level === 0}
      onSelect={setSelectedId}
      selected={selectedId === props.id}
    >
      {getRecursiveMenu(setSelectedId, selectedId, children, level + 1)}
    </CollapseButton>
  ));
}

export function SidebarMenu(props: ISidebarMenuProps): JSX.Element {
  const { menu } = props;
  const [selectedId, setSelectedId] = useState<string | number>();

  return (
    <Paper p="lg" shadow="">
      {getRecursiveMenu(setSelectedId, selectedId, menu)}
    </Paper>
  );
}
