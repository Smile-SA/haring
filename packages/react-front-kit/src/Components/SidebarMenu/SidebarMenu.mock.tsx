import type { IMenuItem } from './types';

import { Chat, HouseLine, Star, User } from '@phosphor-icons/react';

export const menu: IMenuItem[] = [
  {
    children: [{ id: 0, label: 'Home' }],
    id: 1,
    label: 'Home',
    leftIcon: <HouseLine />,
  },
  {
    children: [{ id: 2, label: 'Security' }],
    id: 3,
    label: 'Security',
    leftIcon: <Chat />,
  },
  {
    children: [
      { id: 4, label: 'Release' },
      { children: [{ id: 5, label: 'Account' }], id: 6, label: 'Account' },
      { id: 7, label: 'Upcoming release' },
    ],
    id: 8,
    label: 'Dashboard',
    leftIcon: <Star />,
  },
  { id: 9, label: 'Open Issues', leftIcon: <HouseLine /> },
  {
    children: [
      {
        children: [
          { id: 10, label: 'Wiki pages' },
          { id: 11, label: 'Settings' },
        ],
        id: 12,
        label: 'Dashboard',
      },
      { id: 13, label: 'Home' },
    ],
    id: 14,
    label: 'Pull Requests',
    leftIcon: <User />,
  },
];

export const deeplyNestedMenu: IMenuItem[] = [
  {
    children: [
      {
        children: [
          {
            children: [
              {
                children: [{ id: 16, label: 'Level 4 - A' }],
                id: 8,
                label: 'Level 3 - A',
              },
              {
                children: [{ id: 17, label: 'Level 4 - B' }],
                id: 9,
                label: 'Level 3 - B',
              },
            ],
            id: 4,
            label: 'Level 2 - A',
          },
          {
            children: [
              {
                children: [{ id: 18, label: 'Level 4 - C' }],
                id: 10,
                label: 'Level 3 - C',
              },
              {
                children: [{ id: 19, label: 'Level 4 - D' }],
                id: 11,
                label: 'Level 3 - D',
              },
            ],
            id: 5,
            label: 'Level 2 - B',
          },
        ],
        id: 2,
        label: 'Level 1 - A',
      },
      {
        children: [
          {
            children: [
              {
                children: [{ id: 20, label: 'Level 4 - E' }],
                id: 12,
                label: 'Level 3 - E',
              },
              {
                children: [{ id: 21, label: 'Level 4 - F' }],
                id: 13,
                label: 'Level 3 - F',
              },
            ],
            id: 6,
            label: 'Level 2 - C',
          },
          {
            children: [
              {
                children: [{ id: 22, label: 'Level 4 - G' }],
                id: 14,
                label: 'Level 3 - G',
              },
              {
                children: [{ id: 23, label: 'Level 4 - H' }],
                id: 15,
                label: 'Level 3 - H',
              },
            ],
            id: 7,
            label: 'Level 2 - D',
          },
        ],
        id: 3,
        label: 'Level 1 - B',
      },
    ],
    id: 0,
    label: 'Level 0 - A',
  },
  { children: [{ id: 24, label: 'Level 1 - C' }], id: 1, label: 'Level 0 - B' },
];
