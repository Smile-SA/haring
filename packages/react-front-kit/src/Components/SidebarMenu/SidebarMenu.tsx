'use client';

import type { ICollapseButtonControlledProps } from '../CollapseButton/CollapseButtonControlled';
import type { PaperProps } from '@mantine/core';
import type { ElementType, ReactElement, ReactNode } from 'react';

import { Paper } from '@mantine/core';
import { useMemo, useState } from 'react';

import { addPathAndDepth, flattenNestedObjects } from '../../helpers';
import { CollapseButtonControlled } from '../CollapseButton/CollapseButtonControlled';

export interface IMenuItem<T extends number | string> {
  children?: IMenuItem<T>[];
  id: T;
  label: number | string;
  leftIcon?: ReactNode;
}

type ICollapseButtonProps<T extends number | string, C extends ElementType> =
  | ICollapseButtonControlledProps<T, C>
  | ((item: IMenuItem<T>) => ICollapseButtonControlledProps<T, C>);

export interface ISidebarMenuProps<
  T extends number | string,
  C extends ElementType,
> extends PaperProps {
  /** Props, or function returning props for the CollapseButton component */
  collapseButtonProps?: ICollapseButtonProps<T, C>;
  component?: ElementType;
  /** Default selected menu id */
  defaultSelectedId?: T;
  /** Keeps only one menu per level open at once */
  hasOnlyOneOpenMenu?: boolean;
  /** Menu hierarchy */
  menu: IMenuItem<T>[];
  /** Callback fired when menu is opened or closed */
  onMenuOpen?: (id: T, isOpened: boolean, path: T[]) => void;
  /** Controlled state of which menus are currently open, using `id` field of `IMenuItem` */
  openedMenuIds?: T[];
}

function getRecursiveMenu<T extends number | string, C extends ElementType>(
  setSelectedId: (id?: T) => void,
  onMenuOpen: (id: T, isOpened: boolean) => void,
  openedMenuIds: T[],
  selectedId?: T,
  menu?: IMenuItem<T>[],
  collapseButtonProps?: ICollapseButtonProps<T, C>,
  level = 0,
): ReactElement[] | null {
  if (!menu || menu.length === 0) {
    return null;
  }
  return menu.map((item) => {
    const { children, id, label, leftIcon } = item;
    return (
      <CollapseButtonControlled
        key={id}
        id={id}
        isOpenOnSelect
        label={label}
        leftIcon={leftIcon}
        level={level}
        line={level === 0}
        onCollapseChange={(isOpened) => onMenuOpen(id, isOpened)}
        onSelect={setSelectedId}
        opened={openedMenuIds.includes(id)}
        selected={selectedId === id}
        {...(typeof collapseButtonProps === 'function'
          ? collapseButtonProps(item)
          : collapseButtonProps)}
      >
        {getRecursiveMenu(
          setSelectedId,
          onMenuOpen,
          openedMenuIds,
          selectedId,
          children,
          collapseButtonProps,
          level + 1,
        )}
      </CollapseButtonControlled>
    );
  });
}

/** Additional props will be forwarded to the [Mantine Paper component](https://mantine.dev/core/paper) */
export function SidebarMenu<
  T extends number | string,
  C extends ElementType = 'button',
>(props: ISidebarMenuProps<T, C>): ReactElement {
  const {
    collapseButtonProps,
    component,
    defaultSelectedId,
    hasOnlyOneOpenMenu = false,
    menu,
    onMenuOpen,
    openedMenuIds,
    ...paperProps
  } = props;

  const flatMenu = useMemo(
    () => flattenNestedObjects(addPathAndDepth(menu)),
    [menu],
  );

  const [selectedId, setSelectedId] = useState<T | undefined>(
    defaultSelectedId,
  );
  const [openedIds, setOpenedIds] = useState<T[]>(() => {
    if (defaultSelectedId && !openedMenuIds) {
      const path = flatMenu.find((menu) => menu.id === defaultSelectedId)
        ?.path as T[] | undefined;
      if (path) {
        return path;
      }
    }
    return openedMenuIds ?? [];
  });

  function handleOpenChange(menuId: T, isOpened: boolean): void {
    const openedMenuPath = (flatMenu.find((menu) => menu.id === menuId)?.path ??
      []) as T[];
    onMenuOpen?.(menuId, isOpened, openedMenuPath);
    if (hasOnlyOneOpenMenu && isOpened) {
      setOpenedIds(openedMenuPath);
    } else {
      /** Add or remove id being clicked **/
      const exists = openedIds.includes(menuId);
      let newOpenedIds;
      if (exists) {
        newOpenedIds = openedIds.filter((id) => id !== menuId);
      } else {
        newOpenedIds = openedIds.concat(menuId);
      }
      setOpenedIds(newOpenedIds);
    }
  }

  return (
    <Paper
      // @ts-expect-error wrong type for polymorphic component
      component={component}
      p="lg"
      shadow=""
      {...paperProps}
    >
      {getRecursiveMenu(
        setSelectedId,
        handleOpenChange,
        openedIds,
        selectedId,
        menu,
        collapseButtonProps,
      )}
    </Paper>
  );
}
