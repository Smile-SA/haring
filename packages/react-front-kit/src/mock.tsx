/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
