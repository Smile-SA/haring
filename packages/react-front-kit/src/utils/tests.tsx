import type {
  Queries,
  RenderOptions,
  RenderResult,
  queries,
} from '@testing-library/react';
import type { ReactElement, ReactNode } from 'react';

import { render } from '@testing-library/react';

import { Provider } from '../3-custom/Provider/Provider';

export function renderWithProviders<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container
>(
  ui: ReactElement,
  renderOptions: RenderOptions<Q, Container, BaseElement> = {}
): RenderResult<Q, Container, BaseElement> {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return <Provider>{children}</Provider>;
  }
  return render<Q, Container, BaseElement>(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}
