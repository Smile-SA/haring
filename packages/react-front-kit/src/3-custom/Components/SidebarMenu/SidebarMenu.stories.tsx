import type { Meta, StoryObj } from '@storybook/react';

import { SidebarMenu as Cmp } from './SidebarMenu';
import { menu } from './SidebarMenu.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/SidebarMenu',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const SidebarMenu: IStory = {
  args: {
    menu,
  },
};
