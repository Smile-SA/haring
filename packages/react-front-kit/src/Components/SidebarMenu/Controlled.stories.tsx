import type { Meta, StoryObj } from '@storybook/react';

import { useStorybookArgsConnect } from '@smile/react-front-kit-shared/src/storybook-utils';

import { menu } from './SidebarMenu.mock';
import { SidebarMenuControlled as Cmp } from './SidebarMenuControlled';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onCollapseChange: 'openedMenuIds',
        onSelectedChange: 'selectedId',
      });
      return <Story args={{ ...args }} />;
    },
  ],
  tags: [''],
  title: '3-custom/Components/SidebarMenu',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const Controlled: IStory = {
  args: {
    component: 'nav',
    menu,
    openedMenuIds: [],
  },
};
