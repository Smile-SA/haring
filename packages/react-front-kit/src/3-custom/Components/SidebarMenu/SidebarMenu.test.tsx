import { renderWithProviders } from '../../../utils/tests';

import { SidebarMenu } from './SidebarMenu';
import { menu } from './SidebarMenu.mock';

describe('Header', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(<SidebarMenu menu={menu} />);
    expect(container).toMatchSnapshot();
  });
});
