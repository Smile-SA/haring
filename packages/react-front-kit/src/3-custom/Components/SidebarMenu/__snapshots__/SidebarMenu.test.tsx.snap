// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <div
    class="mantine-Paper-root mantine-fybk5m"
  >
    <div>
      <div
        class="mantine-UnstyledButton-root mantine-Button-root mantine-7ol2s6"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
          >
            <div
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1q46szy"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240,208H224V115.55a16,16,0,0,0-5.17-11.78l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208H16a8,8,0,0,0,0,16H240a8,8,0,0,0,0-16ZM48,115.55l.11-.1L128,40l79.9,75.43.11.1V208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48ZM144,208H112V160h32Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="mantine-Button-label mantine-12tfho7"
          >
            <button
              class="mantine-o4vq62"
              type="button"
            >
              Home
            </button>
          </span>
          <span
            class="mantine-Button-icon mantine-Button-rightIcon mantine-1ydydo3"
          >
            <button
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1v3i31z"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="mantine-19nldqr"
        style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
      >
        <div
          style="opacity: 0; transition: opacity 200ms ease;"
        >
          <div>
            <div
              class="mantine-UnstyledButton-root mantine-Button-root mantine-zap96e"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-Button-label mantine-1nlid6w"
                >
                  <button
                    class="mantine-o4vq62"
                    type="button"
                  >
                    Home
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="mantine-UnstyledButton-root mantine-Button-root mantine-7ol2s6"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
          >
            <div
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1q46szy"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M216,48H40A16,16,0,0,0,24,64V224a15.84,15.84,0,0,0,9.25,14.5A16.05,16.05,0,0,0,40,240a15.89,15.89,0,0,0,10.25-3.78.69.69,0,0,0,.13-.11L82.5,208H216a16,16,0,0,0,16-16V64A16,16,0,0,0,216,48ZM40,224h0ZM216,192H82.5a16,16,0,0,0-10.3,3.75l-.12.11L40,224V64H216Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="mantine-Button-label mantine-12tfho7"
          >
            <button
              class="mantine-o4vq62"
              type="button"
            >
              Security
            </button>
          </span>
          <span
            class="mantine-Button-icon mantine-Button-rightIcon mantine-1ydydo3"
          >
            <button
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1v3i31z"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="mantine-19nldqr"
        style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
      >
        <div
          style="opacity: 0; transition: opacity 200ms ease;"
        >
          <div>
            <div
              class="mantine-UnstyledButton-root mantine-Button-root mantine-zap96e"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-Button-label mantine-1nlid6w"
                >
                  <button
                    class="mantine-o4vq62"
                    type="button"
                  >
                    Security
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="mantine-UnstyledButton-root mantine-Button-root mantine-7ol2s6"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
          >
            <div
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1q46szy"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M239.2,97.29a16,16,0,0,0-13.81-11L166,81.17,142.72,25.81h0a15.95,15.95,0,0,0-29.44,0L90.07,81.17,30.61,86.32a16,16,0,0,0-9.11,28.06L66.61,153.8,53.09,212.34a16,16,0,0,0,23.84,17.34l51-31,51.11,31a16,16,0,0,0,23.84-17.34l-13.51-58.6,45.1-39.36A16,16,0,0,0,239.2,97.29Zm-15.22,5-45.1,39.36a16,16,0,0,0-5.08,15.71L187.35,216v0l-51.07-31a15.9,15.9,0,0,0-16.54,0l-51,31h0L82.2,157.4a16,16,0,0,0-5.08-15.71L32,102.35a.37.37,0,0,1,0-.09l59.44-5.14a16,16,0,0,0,13.35-9.75L128,32.08l23.2,55.29a16,16,0,0,0,13.35,9.75L224,102.26S224,102.32,224,102.33Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="mantine-Button-label mantine-12tfho7"
          >
            <button
              class="mantine-o4vq62"
              type="button"
            >
              Dashboard
            </button>
          </span>
          <span
            class="mantine-Button-icon mantine-Button-rightIcon mantine-1ydydo3"
          >
            <button
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1v3i31z"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="mantine-19nldqr"
        style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
      >
        <div
          style="opacity: 0; transition: opacity 200ms ease;"
        >
          <div>
            <div
              class="mantine-UnstyledButton-root mantine-Button-root mantine-zap96e"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-Button-label mantine-1nlid6w"
                >
                  <button
                    class="mantine-o4vq62"
                    type="button"
                  >
                    Release
                  </button>
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="mantine-UnstyledButton-root mantine-Button-root mantine-j2rc4w"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-Button-label mantine-1nlid6w"
                >
                  <button
                    class="mantine-o4vq62"
                    type="button"
                  >
                    Account
                  </button>
                </span>
                <span
                  class="mantine-Button-icon mantine-Button-rightIcon mantine-1ydydo3"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1v3i31z"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 256 256"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                      />
                    </svg>
                  </button>
                </span>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="mantine-1avyp1d"
              style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
            >
              <div
                style="opacity: 0; transition: opacity 200ms ease;"
              >
                <div>
                  <div
                    class="mantine-UnstyledButton-root mantine-Button-root mantine-vzi7n5"
                    data-button="true"
                    type="button"
                  >
                    <div
                      class="mantine-1wpc1xj mantine-Button-inner"
                    >
                      <span
                        class="mantine-Button-label mantine-ujfmhm"
                      >
                        <button
                          class="mantine-o4vq62"
                          type="button"
                        >
                          Account
                        </button>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="mantine-UnstyledButton-root mantine-Button-root mantine-zap96e"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-Button-label mantine-1nlid6w"
                >
                  <button
                    class="mantine-o4vq62"
                    type="button"
                  >
                    Upcoming release
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="mantine-UnstyledButton-root mantine-Button-root mantine-1r9qeis"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
          >
            <div
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1q46szy"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240,208H224V115.55a16,16,0,0,0-5.17-11.78l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208H16a8,8,0,0,0,0,16H240a8,8,0,0,0,0-16ZM48,115.55l.11-.1L128,40l79.9,75.43.11.1V208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48ZM144,208H112V160h32Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="mantine-Button-label mantine-12tfho7"
          >
            <button
              class="mantine-o4vq62"
              type="button"
            >
              Open Issues
            </button>
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="mantine-UnstyledButton-root mantine-Button-root mantine-7ol2s6"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
          >
            <div
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1q46szy"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="mantine-Button-label mantine-12tfho7"
          >
            <button
              class="mantine-o4vq62"
              type="button"
            >
              Pull Requests
            </button>
          </span>
          <span
            class="mantine-Button-icon mantine-Button-rightIcon mantine-1ydydo3"
          >
            <button
              class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1v3i31z"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="mantine-19nldqr"
        style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
      >
        <div
          style="opacity: 0; transition: opacity 200ms ease;"
        >
          <div>
            <div
              class="mantine-UnstyledButton-root mantine-Button-root mantine-j2rc4w"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-Button-label mantine-1nlid6w"
                >
                  <button
                    class="mantine-o4vq62"
                    type="button"
                  >
                    Dashboard
                  </button>
                </span>
                <span
                  class="mantine-Button-icon mantine-Button-rightIcon mantine-1ydydo3"
                >
                  <button
                    class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1v3i31z"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 256 256"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                      />
                    </svg>
                  </button>
                </span>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="mantine-1avyp1d"
              style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
            >
              <div
                style="opacity: 0; transition: opacity 200ms ease;"
              >
                <div>
                  <div
                    class="mantine-UnstyledButton-root mantine-Button-root mantine-vzi7n5"
                    data-button="true"
                    type="button"
                  >
                    <div
                      class="mantine-1wpc1xj mantine-Button-inner"
                    >
                      <span
                        class="mantine-Button-label mantine-ujfmhm"
                      >
                        <button
                          class="mantine-o4vq62"
                          type="button"
                        >
                          Wiki pages
                        </button>
                      </span>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class="mantine-UnstyledButton-root mantine-Button-root mantine-vzi7n5"
                    data-button="true"
                    type="button"
                  >
                    <div
                      class="mantine-1wpc1xj mantine-Button-inner"
                    >
                      <span
                        class="mantine-Button-label mantine-ujfmhm"
                      >
                        <button
                          class="mantine-o4vq62"
                          type="button"
                        >
                          Settings
                        </button>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="mantine-UnstyledButton-root mantine-Button-root mantine-zap96e"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-Button-label mantine-1nlid6w"
                >
                  <button
                    class="mantine-o4vq62"
                    type="button"
                  >
                    Home
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
